src/ops/factory.rs

```diff
@@ -0,0 +1,901 @@
+    ) -> Result<Array, DataStoreError> {
+        if shape.iter().any(|&i| i < 0) {
+            return Err(DataStoreError::NegativeDimensions(
+                "negative dimensions in shape not allowed".to_string(),
+            ));
+        }
+
+        Ok(unsafe { Self::full_device_unchecked::<T>(shape, values, stream) })
+    }
+
+    /// Create a square identity matrix.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use mlx::{Array, StreamOrDevice};
+    /// //  create [10, 10] array with 1's on the diagonal.
+    /// let r = Array::identity_device::<f32>(10, StreamOrDevice::default());
+    /// ```
+    ///
+    /// # Params
+    ///
+    /// - n: number of rows and columns in the output
+    /// - stream: stream or device to evaluate on
+    #[default_device]
+    pub fn identity_device<T: ArrayElement>(n: i32, stream: StreamOrDevice) -> Array {
+        Self::eye_device::<T>(n, Some(n), None, stream)
+    }
+
+    /// Create a square identity matrix without validating params.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use mlx::{Array, StreamOrDevice};
+    /// //  create [10, 10] array with 1's on the diagonal.
+    /// let r = unsafe { Array::identity_device_unchecked::<f32>(10, StreamOrDevice::default()) };
+    /// ```
+    ///
+    /// # Params
+    ///
+    /// - n: number of rows and columns in the output
+    /// - stream: stream or device to evaluate on
+    pub unsafe fn identity_device_unchecked<T: ArrayElement>(
+        n: i32,
+        stream: StreamOrDevice,
+    ) -> Array {
+        Self::eye_device_unsafe::<T>(n, Some(n), None, stream)
+    }
+
+    /// Create a square identity matrix returning an error if params are invalid.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use mlx::{Array, StreamOrDevice};
+    /// //  create [10, 10] array with 1's on the diagonal.
+    /// let r = Array::try_identity_device::<f32>(10, StreamOrDevice::default()).unwrap();
+    /// ```
+    ///
+    /// # Params
+    ///
+    /// - n: number of rows and columns in the output
+    /// - stream: stream or device to evaluate on
+    pub fn try_identity_device<T: ArrayElement>(
+        n: i32,
+        stream: StreamOrDevice,
+    ) -> Result<Array, DataStoreError> {
+        Self::try_eye_device::<T>(n, Some(n), None, stream)
+    }
+
+    /// Generate `num` evenly spaced numbers over interval `[start, stop]`.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use mlx::{Array, StreamOrDevice};
+    /// // Create a 50 element 1-D array with values from 0 to 50
+    /// let r = Array::linspace_device::<f32, _>(0, 50, None, StreamOrDevice::default());
+    /// ```
+    ///
+    /// # Params
+    ///
+    /// - start: start value
+    /// - stop: stop value
+    /// - count: number of samples
+    /// - stream: stream or device to evaluate on
+    #[default_device]
+    pub fn linspace_device<T, U>(
+        start: U,
+        stop: U,
+        count: Option<i32>,
```

Comment:
Maybe the default count should be documented?

Reply:
This is not a big deal tho, since the user can always refer to the mlx python docs. It's up to you whether this should be documented here